File: src/main/kotlin/com/store/model/ProductDetails.java
package com.store.model;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;

public class ProductDetails {
    @NotNull
    @JsonDeserialize(using = StringOnlyDeserializer.class)
    public String name;

    @NotNull
    public ProductType type;

    @NotNull
    @Min(1)
    public int inventory;

    @NotNull
    @Min(1)
    public double cost;

    public ProductDetails() {}

    public ProductDetails(String name, ProductType type, int inventory, double cost) {
        this.name = name;
        this.type = type;
        this.inventory = inventory;
        this.cost = cost;
    }
}

File: src/main/kotlin/com/store/models/ProductDetails.kt
package com.store.models

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class ProductDetails @JsonCreator constructor(
    @JsonProperty("name") val name: String,
    @JsonProperty("type") val type: String,
    @JsonProperty("inventory") val inventory: Int,
    @JsonProperty("cost") val cost: Float
)

data class ProductId(
    val id: Int
)

data class Product(
    val id: Int,
    val name: String,
    val type: String,
    val inventory: Int,
    val cost: Float
)

data class ErrorResponseBody(
    val timestamp: String,
    val status: Int,
    val error: String,
    val path: String
)